test/req/steps_test.exs

```diff
@@ -97,36 +111,79 @@ defmodule Req.StepsTest do
       end
     end)
 
-    assert Req.get!(c.url <> "/auth").status == 401
+    directory = System.get_env("NETRC", System.user_home!())
 
-    File.write!("#{c.tmp_dir}/empty_netrc", "")
-    assert Req.get!(c.url <> "/auth", netrc: "#{c.tmp_dir}/empty_netrc").status == 401
+    if File.exists?(Path.join(directory, ".netrc")) do
+      assert Req.get!(c.url <> "/auth", auth: :netrc).status == 401
+    else
+      assert_raise RuntimeError, "Error reading .netrc file: no such file or directory", fn ->
+        Req.get!(c.url <> "/auth", auth: :netrc)
+      end
+    end
 
-    File.write!("#{c.tmp_dir}/wrong_netrc", """
+    File.write!("#{c.tmp_dir}/custom_netrc", """
     machine localhost
-    username bad
-    password bad
+    login foo
+    password bar
     """)
 
-    assert Req.get!(c.url <> "/auth", netrc: "#{c.tmp_dir}/wrong_netrc").status == 401
+    assert Req.get!(c.url <> "/auth", auth: {:netrc, c.tmp_dir <> "/custom_netrc"}).status == 200
 
-    File.write!("#{c.tmp_dir}/correct_netrc", """
+    File.write!("#{c.tmp_dir}/wrong_netrc", """
     machine localhost
-    username foo
-    password bar
+    login bad
+    password bad
     """)
 
-    assert Req.get!(c.url <> "/auth", netrc: "#{c.tmp_dir}/correct_netrc").status == 200
+    assert Req.get!(c.url <> "/auth", auth: {:netrc, "#{c.tmp_dir}/wrong_netrc"}).status == 401
+
+    File.write!("#{c.tmp_dir}/empty_netrc", "")
+
+    assert_raise RuntimeError, ".netrc file is empty.", fn ->
+      Req.get!(c.url <> "/auth", auth: {:netrc, "#{c.tmp_dir}/empty_netrc"})
+    end
 
     File.write!("#{c.tmp_dir}/bad_netrc", """
     bad
     """)
 
-    assert_raise RuntimeError, "parse error: \"bad\"", fn ->
-      Req.get!(c.url <> "/auth", netrc: "#{c.tmp_dir}/bad_netrc")
+    assert_raise RuntimeError, "Error parsing .netrc.", fn ->
+      Req.get!(c.url <> "/auth", auth: {:netrc, "#{c.tmp_dir}/bad_netrc"})
     end
   end
 
+  test "encode_headers/1", c do
```

Comment:
thanks for the extra coverage!

Reply:
Sadly that was already there; my changes were so diff-unfriendly that it looks new. :)